back/src/scripts/bin/updateFinalOperations.ts

```diff
@@ -0,0 +1,67 @@
+import { prisma } from "@td/prisma";
+import { Form } from "@prisma/client";
+
+import { closeQueues } from "../../queue/producers";
+import { logger } from "@td/logger";
+import { FINAL_OPERATION_CODES } from "../../common/operationCodes";
+import { operationHooksQueue } from "../../queue/producers/operationHook";
+
+/**
+ * Process all FinalOperations of Forms with a FINAL_OPERATION_CODES
+ * using the async job queue
+ */
+async function exitScript() {
+  logger.info("Done adding updateFinalOperations job the job queue, exiting");
+  await prisma.$disconnect();
+  await closeQueues();
+  process.exit(0);
+}
+
+(async function () {
+  const PAGE_SIZE = 10_000;
+  let pageNumber = 0;
+  let processed = 0;
+  let finalOperationCodeForms: Pick<Form, "id">[] = [];
+  do {
+    finalOperationCodeForms = await prisma.form.findMany({
+      where: {
+        processingOperationDone: {
+          in: FINAL_OPERATION_CODES
+        }
+      },
+      select: {
+        id: true
+      },
+      take: PAGE_SIZE,
+      skip: pageNumber * PAGE_SIZE
```

Comment:
On devrait plutot faire une pagination par curseur ici. En mode offset il doit lire tous les éléments précédents à chaque fois.
cf https://www.prisma.io/docs/orm/prisma-client/queries/pagination#-cons-of-offset-pagination

Reply:
C'est fait désormais, merci.

L'update testé sur une base de prod dure 5 minutes.